ttt/lib/ttt/board.ex

```diff
@@ -12,30 +16,138 @@ defmodule TTT.Board do
     |> mark_with_fewest_moves
   end
 
-  def get_move_count_per_mark(board, marks) do
+  def game_over?(board) do
+    Enum.any?([not remaining_spaces?(board), found_winner?(board)], is_true?)
+  end
+
+  def winning_mark(board) do
+    winning_mark_in_row(board) <>
+     winning_mark_in_column(board) <>
+      winning_mark_in_diagonal(board)
+  end
+
+  def found_winner?(board) do
+    Enum.any?(win_in_any_direction?(board), is_true?)
+  end
+
+  defp win_in_any_direction?(board) do
+    [column_win?(board), row_win?(board), diagonal_win?(board)]
+  end
+
+  defp winning_mark_in_diagonal(board) do
+    diagonal_marks(board)
+    |> winning_mark_in_row
+  end
+
+  defp winning_mark_in_column(board) do
+    transpose(board)
+    |> winning_mark_in_row
+  end
+
+  defp winning_mark_in_row([]), do: ""
+  defp winning_mark_in_row([row | tail]) do
+    if not all_same_mark?([mark | _] = row) do
+      winning_mark_in_row(tail)
+    else
+      mark
+    end
+  end
+
+  def all_same_mark?([mark | t]) do
+    Enum.all?(t, fn(move) -> move == mark end)
+  end
+
+  def row_win?(board) do
+    found_winner_in_row?(board)
+  end
+
+  def column_win?(board) do
+    transpose(board)
+    |> row_win?
+  end
+
+  def diagonal_win?(board) do
+    diagonal_marks(board)
+    |> row_win?
+  end
+
+  defp found_winner_in_row?([], all_same), do: all_same
+  defp found_winner_in_row?(_, true), do: true
+  defp found_winner_in_row?([head | tail], false) do
+    found_winner_in_row?(tail, all_same_mark?(head))
+  end
+  def found_winner_in_row?(board) do
+    found_winner_in_row?(board, false)
+  end
+
+  def diagonal_marks(board) do
+    [diagonal_values(board) , diagonal_values(Enum.reverse(board))]
+  end
+
+  defp diagonal_values(board) do
+    Enum.with_index(board)
+    |> marks_in_diagonal
+  end
+
+  defp marks_in_diagonal(board_with_index) do
+    Enum.map(board_with_index, fn({ row, index }) -> Enum.at(row, index) end)
+  end
+
+  def remaining_spaces?(board) do
+    unique_elements(board)
+    |> Enum.any?(fn(elem) -> is_integer(elem) end)
+  end
+
+  def unique_elements(board) do
+    Enum.uniq(List.flatten(board))
+  end
+
+  defp replace_at(board, move, mark, element, row_index, column_index) do
+    is_position?(board_dimension(board), row_index, column_index, move)
```

Comment:
Break `board_dimension(board)` into its own line to make this function easier to read:
```
board
|> dimension
|> is_position?(row_index, column_index, move)
|> position_value(mark_element)
```
You'll notice that the sequence of 'steps' doesn't flow easily. This may be a sign that these functions need refactoring.